avakas/avakas.py

```diff
@@ -76,24 +77,97 @@ def read(cls):
     def write(cls):
         """Write version data to a project"""
 
-    def bump(self, bump=None):
+    def bump(self,
+             bump=None,
+             prerelease=False,
+             prerelease_prefix=None,
+             build_date=None):
         """Bump version"""
+
+        original = self._version
+
         if not bump:
             return False
 
         if bump == 'patch':
-            self._version = self._version.next_patch()
+            bump_method = self._version.next_patch
         elif bump == 'minor':
-            self._version = self._version.next_minor()
+            bump_method = self._version.next_minor
         elif bump == 'major':
-            self._version = self._version.next_major()
+            bump_method = self._version.next_major
         else:
             raise AvakasError("Invalid version component")
 
+        self._version = bump_method()
+
+        if prerelease:
+            prerelease_version = self._get_prerelease_version(
+                original, prefix=prerelease_prefix
+            )
+
+            self.apply_prerelease(prerelease_version,
+                                  prefix=prerelease_prefix,
+                                  build_date=build_date)
+
+        assert self._version != original
```

Comment:
`assert`s are removed when python is run with [-O](https://docs.python.org/3/using/cmdline.html#cmdoption-o). Suggest changing this to throw an exception.

Reply:
This was testing code that I didnt' intend to submit,  making this throw an exception (and handling it) feels conceptually like "another PR" but I don't know if that's me having a foolish consistency problem